### ISSUE TITLE
Running a code chunk in R script file does not work in latest RStudio preview - Windows 10

### ISSUE THREAD

**Issue by djmurphy420** [rstudio/rstudio#7163]
I upgraded to R-4.0.1 yesterday and installed the most recent RStudio preview for Windows 10. I have encountered a couple of buglets that I'd like to pass along.

1. I am unable to highlight a code chunk in an R script file and run it by clicking on the Run button. Of course, copying and pasting the code into the console still works.

2. When I try to View an R data object, an error box pops up with the message: 

Show Data Frame:
Exception caught: (Type Error): Cannot read property 'substr' of null

When I click OK in the box, it disappears and the file appears as expected. 

No lasting damage, but the behavior is not consistent with past experience in RStudio preview.

**Comment by kevinushey** [comment]
Looks like this exception is caused by a null `extendedType_`. (obfuscated GWT stack but at least that's clear from the parameters)

<img width="691" alt="Screen Shot 2020-06-18 at 12 39 12 PM" src="https://user-images.githubusercontent.com/1976582/85064682-c21eda00-b160-11ea-9bf2-1686db390258.png">

I think we need null checks in the places where we check `XT_RMARKDOWN_PREFIX`; e.g.

```
$ rg XT_RMARKDOWN_PREFIX
src/gwt/src/org/rstudio/studio/client/workbench/views/source/Source.java
3975:               activeEditor_.getExtendedFileType().startsWith(SourceDocument.XT_RMARKDOWN_PREFIX) ||
4006:            activeEditor_.getExtendedFileType().startsWith(SourceDocument.XT_RMARKDOWN_PREFIX);

src/gwt/src/org/rstudio/studio/client/workbench/views/source/model/SourceDocument.java
187:   public final static String XT_RMARKDOWN_PREFIX = "rmarkdown-";

src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTarget.java
1704:      if (extendedType_.startsWith(SourceDocument.XT_RMARKDOWN_PREFIX))
2914:      if (extendedType.startsWith(SourceDocument.XT_RMARKDOWN_PREFIX))
5604:      if (extendedType.startsWith(SourceDocument.XT_RMARKDOWN_PREFIX))

src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTargetWidget.java
1251:         else if (type.startsWith(SourceDocument.XT_RMARKDOWN_PREFIX))
```

cc: @jmcphers as I think you touched this most recently.

**Comment by dataning** [comment]
I think I am having a similar issue but in Mac. 

So if I import a CSV into RStudio, it would pop this `Exception caught: (Type Error): Cannot read property 'substr' of null`. 

But that's not the biggest headache. It's the "non-responsive" behaviour - hanging/lagging during saving files or executing a line of code. I checked two Macs and they seem to suffer similar problems. Never experienced this in Preview version before.

**Comment by jmcphers** [comment]
This was due to a bad merge from 1.4 to 1.3, and has been fixed with https://github.com/rstudio/rstudio/commit/f1c6313f03591eb0351505e90776c43a83df8256. Version 1.3.1039-1 should work correctly:

https://rstudio.com/products/rstudio/download/preview/

**Comment by dataning** [comment]
@jmcphers Still, the same. I downloaded the latest Preview Version 1.3.1039-1 and dragged it into my Application (Mac). I did it twice - just in case. 

The ongoing problem: If I open a new R script, write something down, and try to give a file name and save it, it will "hang" every time when I press "Save". I cannot do anything but "Force to quit". I'm just wondering if I should uninstall previous RStudio Preview before dragging the new version in?

**Comment by kevinushey** [comment]
> The ongoing problem: If I open a new R script, write something down, and try to give a file name and save it, it will "hang" every time when I press "Save". I cannot do anything but "Force to quit". I'm just wondering if I should uninstall previous RStudio Preview before dragging the new version in?

I'm not able to reproduce this issue, unfortunately.

**Comment by R-Hannibal** [comment]
@dataning - Are you still having the ongoing issues you mentioned in your previous comment? 

We have a fix in place for this ticket that I can confirm IS in our latest preview build 1.3.1054 (available [HERE](https://rstudio.com/products/rstudio/download/preview/)). If the issues you mentioned are persisting in this build we may need to take another look, either way please let us know!

**Comment by R-Hannibal** [comment]
Verified Fixed on: RStudio Version - 1.3.1054, R Version - 4.0.2, Windows 10 x64 - Desktop

This looks like it is fixed in the latest preview. Focused testing on the `Exception caught: (Type Error): Cannot read property 'substr' of null` error being produced on data import symptom for issue reproduction and verification. Can confirm that this error was occurring in 1.3.1033 and no longer occurs in 1.3.1054.

@dataning we were never able to reproduce the additional issue you mentioned in an earlier comment (IDE hanging after attempting to save a file). If that issue is still occurring for you please let us know/open a new ticket so that we can take a look at it. Thank you for helping us to improve RStudio.

**Comment by djmurphy420** [comment]
I can confirm that my issues were resolved starting with the June 27 Windows release. Thank you!!

Best regards,
Dennis
     On Monday, July 6, 2020, 09:11:52 AM PDT, Mario Ferrini <notifications@github.com> wrote:  
 
 


Verified Fixed on: RStudio Version - 1.3.1054, R Version - 4.0.2, Windows 10 x64 - Desktop

This looks like it is fixed in the latest preview. Focused testing on the Exception caught: (Type Error): Cannot read property 'substr' of null error being produced on data import symptom for issue reproduction and verification. Can confirm that this error was occurring in 1.3.1033 and no longer occurs in 1.3.1054.

@dataning we were never able to reproduce the additional issue you mentioned in an earlier comment (IDE hanging after attempting to save a file). If that issue is still occurring for you please let us know/open a new ticket so that we can take a look at it. Thank you for helping us to improve RStudio.
